File: backend/go/services/notifications.go
package services

import (
	"bytes"
	"encoding/json"
	"net/http"
)

func QueueEmailNotification(userID, eventType string, payload interface{}) {
	body, err := json.Marshal(map[string]interface{}{
		"user_id":    userID,
		"event_type": eventType,
		"payload":    payload,
	})
	if err != nil {
		return
	}

	http.Post("http://localhost:8000/internal/notifications/queue", "application/json", bytes.NewReader(body))
}

File: backend/python/routes/notifications.py
from flask import request, jsonify

from services.email_service import send_email


def register_routes(app):
    @app.route("/health")
    def health():
        return "ok"

    @app.route("/internal/notifications/queue", methods=["POST"])
    def queue_notification():
        data = request.get_json()

        user_id = data.get("user_id")
        event_type = data.get("event_type")
        payload = data.get("payload")

        print(f"[notification] to user {user_id}: {event_type} — {payload}")

        to_address = payload.get("email", "unknown@example.com")
        subject = "SellEasy notification"
        body = f"You have a new SellEasy notification: {event_type} — {payload}"

        send_email(to_address, subject, body)

        return jsonify({"queued": True})
